src/tests/gtests.cpp

```diff
@@ -174,3 +174,74 @@ TEST_F(LeelaTest, MoveOnOccupiedSq) {
     // Find this error in the output
     EXPECT_NE(output.find("illegal move"), std::string::npos);
 }
+
+
+TEST_F(LeelaTest, TimeControl) {
+    // Initialize network
+    cfg_weightsfile = "../src/tests/0k.txt";
+    Network::initialize();
+    cfg_max_playouts = 1;
+    cfg_num_threads = 1;
+    std::string output;
+
+    // TODO I tried to make these separate tests,
+    // but it died locally when leelaz ran the second time.
+    {
+        auto maingame = get_gamestate();
+        // clear_board to force GTP to make a new UCTSearch.
+        // This will pickup our new cfg_* settings.
+        GTP::execute(maingame, "clear_board");
+
+        GTP::execute(maingame, "kgs-time_settings byoyomi 0 100 1");
+        GTP::execute(maingame, "go");
+        testing::internal::CaptureStderr();
+        GTP::execute(maingame, "showboard");
+        output = testing::internal::GetCapturedStderr();
+        EXPECT_NE(output.find("Black time: 00:01:40, 1 period(s) of 100 seconds left"), std::string::npos);
+        EXPECT_NE(output.find("White time: 00:01:40, 1 period(s) of 100 seconds left"), std::string::npos);
+
+        GTP::execute(maingame, "kgs-time_settings byoyomi 0 120 1");
+        GTP::execute(maingame, "go");
+        testing::internal::CaptureStderr();
+        GTP::execute(maingame, "showboard");
+        output = testing::internal::GetCapturedStderr();
+        EXPECT_NE(output.find("Black time: 00:02:00, 1 period(s) of 120 seconds left"), std::string::npos);
+        EXPECT_NE(output.find("White time: 00:02:00, 1 period(s) of 120 seconds left"), std::string::npos);
+    }
+
+    {
+        cfg_max_playouts = 0;
+        auto maingame = get_gamestate();
+        // clear_board to force GTP to make a new UCTSearch.
+        // This will pickup our new cfg_* settings.
+        GTP::execute(maingame, "clear_board");
+
+        // Absolute time 100s = 1.37s per move in opening.
+        // Enough to be visible on display_times
+        GTP::execute(maingame, "time_settings 100 0 0");
+        // Use assert here because if the time breaks we might
+        // cause the test to think forever.
+        testing::internal::CaptureStderr();
+        GTP::execute(maingame, "showboard");
+        output = testing::internal::GetCapturedStderr();
+        printf("output\n%s\noutput", output.c_str());
+        EXPECT_NE(output.find("Black time: 00:01:40"), std::string::npos);
+        EXPECT_NE(output.find("White time: 00:01:40"), std::string::npos);
+
+        GTP::execute(maingame, "go");
+        testing::internal::CaptureStderr();
+        GTP::execute(maingame, "showboard");
+        output = testing::internal::GetCapturedStderr();
+        printf("output\n%s\noutput", output.c_str());
+        EXPECT_NE(output.find("Black time: 00:01:38"), std::string::npos);
```

Comment:
Pff, what if the VM this inevitably runs on in CI is slow to respond?

I'd just check if the clock has moved from the starting value, not the exact new one.

Reply:
Using Canadian method instead now.